edx_api/client.py

```diff
@@ -24,51 +24,44 @@ def __init__(self, credentials, base_url='https://courses.edx.org/'):
 
         self.base_url = base_url
         self.credentials = credentials
+        self.timeout = timeout
 
+    @property
     def get_requester(self):
-        """
-        Returns an object to make authenticated requests. See python `requests` for
-        the API.
-        """
-        # TODO(abrahms): Perhaps pull this out into a factory function for
-        # generating an EdxApi instance with the proper requester & credentials.
-        session = requests.session()
-        session.headers.update({
-            'Authorization': 'Bearer {}'.format(self.credentials['access_token'])
-        })
-        return session
+        """Returns an Requester object to make authenticated requests"""
+        return Requester(self.timeout, self.credentials['access_token'])
```

Comment:
We should keep the other methods available, not just `get` and `post`. If you patch `session.request` it should be enough to handle the timeout appropriately, since `get` and `post` and other methods will call `request`.